lib/ecto/changeset.ex

```diff
@@ -940,6 +940,38 @@ defmodule Ecto.Changeset do
     end
   end
 
+
+  @doc """
+  Validates that the given field matches the virtual confirmation of that field.
+
+  The field has to have a virtual field that matches `field_confirmation`.
+  e.g.
+  ``` elixir
+  schema "users" do
+    field :email
+    field :email_confirmation, :string, virtual: true
+  end
+  ```
+
+  ## Options
+
+    * `:message` - the message on failure, defaults to "does not match"
+
+  ## Examples
+
+      validate_confirmation(changeset, :email)
+      validate_confirmation(changeset, :password, message: "passwords do not match")
+
+  """
+  @spec validate_confirmation(t, atom, Enum.t) :: t
+  def validate_confirmation(changeset, field, opts \\ []) do
```

Comment:
We should not require an `email_confirmation` field. Instead, we can simply look inside the params map.